### PR TITLE
Floating average

### LINKED ISSUE #50: Smooth analog potentiometer readings
Copied from https://github.com/MobiFlight/MobiFlight-Connector/issues/519

> **Is your feature request related to a problem? Please describe.**
> I have encountered some electrical noise coming from 7 segment display modules, which cause jitter problems in a potentiometer. Suggestions received have reduced the problem but not eliminated it. It would be good to have a way to smoothen potentiometer readings through software.
> 
> **Describe the solution you'd like**
> Implement a 10 reading rolling average in the arduino firmware with a soft switch in MF Connector analog input configuration, that can be used to turn this feature on and off. When on, the analog pot reading sent from the board will be the rolling average of the last 10 readings. This will effectively smoothen the readings and eliminate jitter. This technique has been successfully used to smoothen the operation of a yoke with three analog pots connected to a pro micro.
> 
> **Describe alternatives you've considered**
> I have tried to isolate the noise problem in hardware and jitter has been successfully reduced, but not eliminated.
> 
> **Additional context**
> Potentiometers can present jitter due to various reasons. This will provide a solution in software to alleviate this problem.
> I suggest that instead of calculating a real rolling average, which can take too much computational power, adding 10 readings and dividing by 10 in each read cycle, Start with an assumed average of zero and an array of ten readings (all zero). In each cycle: increment the reading counter index (0 to 9), add the current reading divided by 10 and substract the counter indexed reading divided by 10. Finally, store the current reading in the counter indexed position (replacing the value just substracted).
> Any error at start disappears within 10 cycles.

### PR DESCRIPTION
**Pull request by elral**
Flaoting average for smoothing the analog input. Sample time is 10ms, floating average is calculated from 8 values. Maybe this must be optimized.

Fixes  #50

### CODE REVIEW

File: _Boards/Atmel/Board_ProMicro/MFBoards.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 #define MF_SEGMENT_SUPPORT 1
 #define MF_LCD_SUPPORT 1
-#define MF_STEPPER_SUPPORT 1
+#define MF_STEPPER_SUPPORT 0
```
Comment (neilenns) [inline _Boards/Atmel/Board_ProMicro/MFBoards.h:6]: Looks like this was an unrelated change accidentally included in this PR?

File: src/MF_Modules/MFAnalog.cpp [hunk 2]
```diff
@@ -17,16 +17,35 @@ MFAnalog::MFAnalog(uint8_t pin, analogEvent callback, const char * name, uint8_t
...
 
 void MFAnalog::update()
-{    
+{
+/* *****************************************************************************
+*   There is a pull request to transfer the millis() to the mobiflight.cpp
+*   this will save some memory and would be also valid for tick();
+********************************************************************************/
     uint32_t now = millis();
+    if (now-_lastTick > 10) {
```
Comment (neilenns) [inline src/MF_Modules/MFAnalog.cpp:26]: When I saw `tick` I thought this was related to clock stuff. How about calling it something like `readPin()` or `readValue()`, and have `_lastTick` be `_lastReadTime`? Just makes it clearer what the function and variable do.

File: include/MFAnalog.h [hunk 3]
```diff
@@ -28,7 +30,8 @@ class MFAnalog
 {
 public:
     MFAnalog(uint8_t pin = 1, analogEvent callback = NULL, const char * name = "Analog Input", uint8_t sensitivity = 2);
-    void update();    
+    void update();
+    void tick(void);
```
Comment (neilenns) [inline include/MFAnalog.h:34]: To keep consistent with the rest of the code this can omit `void` in the function parameters.

File: include/MFAnalog.h [hunk 4]
```diff
@@ -37,5 +40,10 @@ class MFAnalog
     uint32_t      _last;
     analogEvent   _handler; 
     uint8_t       _sensitivity;
+
+    uint16_t ADC_Buffer[ADC_MAX_AVERAGE] = {0};     // Buffer for all values from each channel  
+    uint16_t ADC_Average_Buffer = 0;                // sum of sampled values, must be divided by ADC_MAX_AVERAGE to get actual value
```
Comment (neilenns) [inline include/MFAnalog.h:45]: How about calling it `ADC_Running_Total`?

File: src/MF_Modules/MFAnalog.cpp [hunk 5]
```diff
@@ -17,16 +17,35 @@ MFAnalog::MFAnalog(uint8_t pin, analogEvent callback, const char * name, uint8_t
...
+      tick();
+      _lastTick = millis();
+    }
     if (now-_last <= 50) return; // Analog is too spammy on the protocol to MF otherwise.
+/* *************************************************************************** */
 
-    int newValue = (int) analogRead(_pin);
+    int newValue = ADC_Average_Buffer/ADC_MAX_AVERAGE;
     _last = now;
+//Serial.print("Analog: "); Serial.println(newValue);
```
Comment (neilenns) [inline src/MF_Modules/MFAnalog.cpp:35]: I assume this was just for testing purposes and should be removed from the final checked-in code?

File: include/MFAnalog.h (hunk 3 above)
Comment (elral) [reply include/MFAnalog.h:34]: Yes, changed it.

File: src/MF_Modules/MFAnalog.cpp (hunk 5 above)
Comment (elral) [reply src/MF_Modules/MFAnalog.cpp:35]: Yes, deleted it.

File: src/MF_Modules/MFAnalog.cpp (hunk 2 above)
Comment (neilenns) [reply src/MF_Modules/MFAnalog.cpp:26]: I like `readBuffer()`, good suggestion!

File: include/MFAnalog.h (hunk 4 above)
Comment (neilenns) [reply include/MFAnalog.h:45]: Yeah. Don't know why it selected three lines. Weird.

File: src/MF_Modules/MFAnalog.cpp (hunk 2 above)
Comment (elral) [reply src/MF_Modules/MFAnalog.cpp]: It's changed

File: include/MFAnalog.h (hunk 4 above)
Comment (elral) [reply include/MFAnalog.h:45]: What would be about a combination: `ADC_Average_Total`

Comment (elral) [reply include/MFAnalog.h]: Done.

**Review by neilenns — APPROVED** [review]
